spec/system/admin/orders/having_producer_products/hub_actions_spec.rb

```diff
@@ -0,0 +1,150 @@
+# frozen_string_literal: true
+
+require 'system_helper'
+
+RSpec.describe 'As a hub (producer seller) who have the ability to update orders having their products' do
+  include AdminHelper
+  include AuthenticationHelper
+  include WebHelper
+
+  let!(:hub1) { create(:distributor_enterprise, name: 'My hub1') }
+  let!(:hub1_v1) { create(:variant, supplier_id: hub1.id) }
+  let!(:hub1_v2) { create(:variant, supplier_id: hub1.id) }
+  let(:order_cycle) do
+    create(:simple_order_cycle, distributors: [distributor], variants: [hub1_v1, hub1_v2], coordinator: distributor)
+  end
+  let!(:order_containing_hub1_products) do
+    o = create(
+      :completed_order_with_totals,
+      distributor:, order_cycle:,
+      line_items_count: 1
+    )
+    o.line_items.first.update_columns(variant_id: hub1_v1.id)
+    o
+  end
+  let(:hub1_ent_user) { create(:user, enterprises: [hub1]) }
+
+  context "As hub1 enterprise user" do
+    before { login_as(hub1_ent_user) }
+    let(:order) { order_containing_hub1_products }
+    let(:user) { hub1_ent_user }
+
+    describe 'orders index page' do
+      before { visit spree.admin_orders_path }
+
+      context "when no distributor allow the producer to edit orders" do
+        let(:distributor) { create(:distributor_enterprise) }
+
+        it "should not allow producer to view orders page" do
```

Comment:
This is a bit of a nit pick, but the rspec style guide discourage the use of "should" in example description : https://rspec.rubystyle.guide/#should-in-example-docstrings